src/CountryCode.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\I18n\PhoneNumber;
+
+use Laminas\I18n\PhoneNumber\Exception\InvalidArgumentException;
+use Locale;
+
+use function assert;
+use function preg_match;
+use function strtoupper;
+
+/**
+ * @psalm-immutable
+ */
+final class CountryCode
```

Comment:
Should this rather be placed in laminas-i18n? I think the goal should be the same handling for all i18n components. This means that in the future we will need the value object also in laminas-i18n.

Reply:
At the moment, `i18n` is not a dependency so moving it there would make it one. Also, it's got some `8.1` specific syntax which isn't a drama to downgrade, but that'd be additional discussion - i.e. we could also bump `i18n` to `8.1` minimum (?) - cons aside, the value object would surely be useful and I think you're right that something this basic would be better placed in `i18n`

Reply:
> we could also bump `i18n` to `8.1` minimum (?)

Or 8.0? (Version 7 is as nearly dead.)

Reply:
Happy to send in a PR to `i18n` for this @froschdesign - Shall I just do it, or do you think we need more discussion on the decision?

Reply:
Version 7.4 has already been dropped in other packages, so there is no need for a deeper discussion on this topic.
Moving the value object to hold and detect a locale is also needed in laminas-i18n or will be used there in the future. It is easier to use.
And I don't see it as a problem that a dependency is created, because I would have expected it anyway. 😄

Reply:
I can make the relevant changes here if this is merged: https://github.com/laminas/laminas-i18n/pull/84